extensions/azurePublish/src/node/index.ts

```diff
@@ -0,0 +1,631 @@
+    /* These methods provision resources to azure async */
+    /*******************************************************************************************************************************/
+    asyncProvision = async (jobId: string, config: ProvisionConfig, project: IBotProject, user): Promise<void> => {
+      const { subscription, name } = config;
+      // Create the object responsible for actually taking the provision actions.
+      const azureProvisioner = new BotProjectProvision({
+        ...config,
+        logger: (msg: any) => {
+          this.logger(msg);
+          BackgroundProcessManager.updateProcess(jobId, 202, msg.message);
+        },
+        tenantId: subscription.tenantId, // does the tenantId ever come back from the subscription API we use? it does not appear in my tests.
+      });
+
+      // perform the provision using azureProvisioner.create.
+      // this will start the process, then return.
+      // However, the process will continue in the background
+      try {
+        const provisionResults = await azureProvisioner.create(config);
+        // GOT PROVISION RESULTS!
+        // cast this into the right form for a publish profile
+        const publishProfile = {
+          name: config.hostname,
+          environment: 'composer',
+          hostname: config.hostname,
+          luisResource: `${config.hostname}-luis`,
+          runtimeIdentifier: 'win-x64',
+          settings: {
+            applicationInsights: {
+              InstrumentationKey: provisionResults.appInsights?.instrumentationKey,
+            },
+            cosmosDb: provisionResults.cosmosDB,
+            blobStorage: provisionResults.blobStorage,
+            luis: {
+              authoringKey: provisionResults.luisAuthoring?.authoringKey,
+              authoringEndpoint: provisionResults.luisAuthoring?.authoringEndpoint,
+              endpointKey: provisionResults.luisPrediction?.endpointKey,
+              endpoint: provisionResults.luisPrediction?.endpoint,
+              region: provisionResults.resourceGroup.location,
+            },
+            MicrosoftAppId: provisionResults.appId,
+            MicrosoftAppPassword: provisionResults.appPassword,
+          },
+        };
+
+        console.log(publishProfile);
+
+        // write this to the project settings. Browser need to refresh to get the update
+        // project.settings.publishTargets.push({
+        //   name: config.name,
+        //   type: 'azurePublish',
+        //   configuration: JSON.stringify(publishProfile),
+        //   lastPublished: null,
+        // });
+
+        // await project.updateDefaultSlotEnvSettings(project.settings);
+
+        BackgroundProcessManager.updateProcess(jobId, 200, 'Provision completed successfully!', publishProfile);
+      } catch (error) {
+        BackgroundProcessManager.updateProcess(jobId, 500,
+        `${stringifyError(error)}
+        detail message can see ${ProvisionLog} in your bot folder`);
```

Comment:
Rather than use the node environment to generate this file, use the publishing profile name for the log file.   

So, use the `name` variable here to define it -- provision.[$name].log